### PR TITLE
Give us your best Inscription examples! will be added to documentation for Issue 949

### PR DESCRIPTION
**Pull request by cryptoni9n**
The purpose of this PR is to gather good examples for Issue #949 

Please submit feedback of any on-chain inscriptions that you feel make exemplary use of following inscription technologies for their respective areas of documentation:

3. [Inscriptions](https://docs.ordinals.com/inscriptions.html)

    [3.1. Delegate](https://docs.ordinals.com/inscriptions/delegate.html)
    [3.2. Metadata](https://docs.ordinals.com/inscriptions/metadata.html)
    [3.3. Pointer](https://docs.ordinals.com/inscriptions/pointer.html)
    [3.4. Provenance](https://docs.ordinals.com/inscriptions/provenance.html)
    [3.5. Recursion](https://docs.ordinals.com/inscriptions/recursion.html)

      - a) Remixing the content of existing inscriptions.
      - b) Publishing snippets of code, images, audio, or stylesheets as shared public resources.
      - c) Generative art collections where an algorithm is inscribed as JavaScript, and instantiated from multiple inscriptions with unique seeds.
      - d) Generative profile picture collections where accessories and attributes are inscribed as individual images, or in a shared texture atlas, and then combined, collage-style, in unique combinations in multiple inscriptions.

    [3.6. Rendering](https://docs.ordinals.com/inscriptions/rendering.html)

Updating the list below with all submissions:
------

### Delegation: 
- [This inscription](http://ordinals.com/inscription/23a8f17fff4a73e2932dfc76e46d14d4f8975da96f5d5ae9a45898422056071ai0) was inscribed as a delegate of [this inscription](https://ordinals.com/inscription/9ff39db4c51f831225d41efbd29a399f2b16c758970ec4ab95a1a17e8be59088i0) and is also the parent inscription of a rune.
- The collection [Oscillations *](https://gamma.io/ordinals/collections/oscillations-mdv) utilizes delegate, provenance, recursion, sat-endpoint, and detects the kind of sat that each piece is inscribed on (sattribute-aware).   Each piece is a delegate of [this inscription](https://ordinals.com/inscription/52b4ea10c2518c954c73594e403ccfb2d50044f5a3b09a224dfa3bf06dd1d499i0).

### Metadata: 
- [Metadata to define attributes](http://ordinals.com/inscription/370b02c48b6ce0046445e0d60a9cb0e187bfc9cb093cffcc8c9bd37ce95367cfi0)
- [Metadata feeds inscription itself](https://ordinals.com/inscription/454700af2cffd3a872daeb89598a891f5fe9936225838a8c4522f491f246ca0ai0)

### Provenance: 
- [Bitcoin Magazine Cover](https://ordinals.com/inscription/60902330)
- [Inscription 0](http://ordinals.com/inscription/0)
- [Inscription 9776](http://ordinals.com/inscription/17541f6adf6eb160d52bc6eb0a3546c7c1d2adfe607b1a3cddc72cc0619526adi0)
- With the [Rug Me](https://ordinals.com/inscription/f87a6b16a1224b65f9c8ea55e1718585b3075373581689e343860c12a70d15c7i41) collection, owners can change the background color by inscribing a child to it.

### Recursion
- The [Abstractii Evolved](https://ordinals.com/inscription/c56795a454a30bb6866686770f14d015d53f3cf2ddfa5154e34a7cd1120a51efi0) generative collection uses the recursive blockheight endpoint as a seed 
- (b) [This code](http://ordinals.com/inscription/15838225) is called recursively in [this inscription](http://ordinals.com/inscription/60445754)
- (b) [This code](https://ordinals.com/content/e48af8aebe608656a3739393d1270d88285ab3051fb800743509d82bcf163623i0) is called recursively in [this inscription](https://ordinals.com/inscription/70940369) allowing it to operate as a drawing program.
- (b&c) [Blob](https://ordinals.com/inscription/b10b00bfb146fee3e86d6cbd8e8c954c485da41be480c0b21a6a63de7986892bi0) is a generative collection that seeds generation with `metadata`. It also uses `delegation` as the rendering engine. The Delegate is updatable with the `sat endpoint`. Delegate uses threeJS, React 3 Fiber and other libraries via `recursion`. Also each blob is on a `satname` starting with blob
- (c) [Abstractii Genesis](https://ordinals.com/inscription/b2de70a5658d3659b74a9a349e02ed9396318eab844073b6f35e726dea9e103ei0) uses the inscriptionID as a seed
- (d) [OnChainMonkey](http://ordinals.com/children/fb162a46943e5d7d31d72ee2c8c850e66c1ca5d0d453068aa63883528285ed21i0)

### Rendering:

### CODE REVIEW

**Comment by phorkish** [comment]
Here's an inscription that I made that uses recursion, delegation and metadata. And also provenance if you count the structure of its dependencies.

https://ordinals.com/inscription/4fb16cb8362e117f671fdbc105d6738887cd58432d25cec328212e217426f4fdi0

Each inscription is a full application that reads in the metadata and uses it to draw pixel art. The idea is that you can delegate to this inscription with your own metadata and it'll render your pixel art. If you hover over the inscription there's a menu in the top right, which opens up a drawing tool that generates the metadata that you need to inscribe.

For example, I redrew my Bandit here 

https://ordinals.com/inscription/454700af2cffd3a872daeb89598a891f5fe9936225838a8c4522f491f246ca0ai0

**Comment by elocremarc** [comment]
### Bitcoin Magazine Cover
Child endpoint where the children are rendered into the parent inscription.
https://ordinals.com/inscription/60902330

### Blob 
https://ordinals.com/inscription/b10b00bfb146fee3e86d6cbd8e8c954c485da41be480c0b21a6a63de7986892bi0
Generative collection that seeds generation with `metadata` It also uses `delegation` as the rendering engine. The `Delegate` is updatable with the `sat endpoint`. Delegate uses threeJS, React 3 Fiber and other libraries via `recursion`. Also each blob is on a `satname` starting with blob

Sat endpoint:
https://ordinals.com/sat/1749358685368819

Delegate:
https://ordinals.com/inscription/bc6b4132d7772e78eb4b8390ebe9970c6e5315ec0e0f644a88b4e5b35ced43d3i0

Parent:
https://ordinals.com/inscription/648f02fbb36d7841dbf629966ea9c82a60255044fbdd09b31533c0b9fafa573di0

### Abstractii Genesis
A generative collection that uses inscriptionId as seed
https://ordinals.com/inscription/b2de70a5658d3659b74a9a349e02ed9396318eab844073b6f35e726dea9e103ei0

### Abstractii Evolved
A generative collection that uses `blockheight` as a seed
https://ordinals.com/inscription/c56795a454a30bb6866686770f14d015d53f3cf2ddfa5154e34a7cd1120a51efi0

### Rug Me
Child endpoint and Sat endpoint example collection
Collector can update background color by inscribing a child
Creator can "rug" collection by inscribing an additional image onto the entire collection using the sat endpoint
https://ordinals.com/inscription/f87a6b16a1224b65f9c8ea55e1718585b3075373581689e343860c12a70d15c7i41

**Comment by cryptoni9n** [comment]
thanks @phorkish & @elocremarc for the excellent examples!  I've incorporated them into the list in the OP.

**Comment by matthewvollmer** [comment]
### **Oscillations*
My collection Oscillations * utilizes `delegate`, `provenance`, `recursion`, `sat-endpoint`, and detects the kind of sat that each piece is inscribed on (`sattribute-aware`). 

- Each piece is a **delegate** of this inscription: https://ordinals.com/inscription/52b4ea10c2518c954c73594e403ccfb2d50044f5a3b09a224dfa3bf06dd1d499i0
- The **parent** of this piece is the shared parent across my collections, demonstrating creator provenance. https://ordinals.com/inscription/303ffd160b94411bdae3b53bdc3d5329f912a81e459490c6da45358d7f4802e3i0
- This inscription just renders an iframe. It uses **sat-endpoint** to point to the sat which holds the actual generative code, the "generative engine" https://ordinals.com/inscription/ecb4d0ac876dc53b0b75a2d5f86cf05d41acbe9ba35c711f8a50304734505129i0
- This generative engine uses **recursion** to pull a variety of code libraries (THREE.js & addons), CSS styles, and a text file listing all the Pizza sat ranges. 
- The delegate passes in each piece's inscription ID via a **query paramater**, and the generative engine uses that ID to fetch the inscription info via `/r/inscription/<iid>`. It then uses this info to determine the **sattribute**, based on **sat number** and **charms**
- The inscription ID is also used as a randomization seed for the engine's parameters. 

Collection Link: https://gamma.io/ordinals/collections/oscillations-mdv
The 500 different pieces are on 7 different types of sats, each with different generative engine parameters.

**Comment by cryptoni9n** [comment]
thanks matthewvollmer, I've added your examples to the list in the OP!  I think I bought some runes that are associated with this collection, RUNIC•SUBSTANCE?

**Comment by cryptoni9n** [comment]
seems like we may have all of the suggestions we're going to get so I'll start working on trying to integrate these examples into the docs.

**Comment by cryptoni9n** [comment]
Hi @casey & @raphjaph - I've incorporated the suggestions we received here into the docs.  Please review and let me know if there are any edits or changes that you'd prefer.

**Comment by lifo** [comment]
Spellbound

https://ordinals.com/children/451accbce30177c23a8cd42ab85131312a449359ada3c054aa147b054136fa3bi0

Utilizes recursion, delegation, metadata, parent/child, postage, location, compression. Background here https://x.com/lifofifo/status/1767059753555853656

**Comment by lifofifoX** [comment]
Spellbound

https://ordinals.com/children/451accbce30177c23a8cd42ab85131312a449359ada3c054aa147b054136fa3bi0

Utilizes recursion, delegation, metadata, parent/child, postage, location, compression. Background here https://x.com/lifofifo/status/1767059753555853656

**Comment by DrJingLee** [comment]
Actully GPU Ordinals by Anarkoic is another cool collection ,here are some self-intro from the Magic Eden: 

GPU Ordinals by Anarkoic is the first ever WebGPU Ordinals collection. WebGPU is the new technology that provides high-throughput use of GPU's for graphics and even computation, & can enable high-performance Metaverses and LLM AI's on Bitcoin. This immersive, interactive 3D collection also features Grandparent-Grandchild Inscription, Spatial Audio, Real-time Block Streaming & introduces Super-Recursion. Max Supply 2100.

https://magiceden.io/ordinals/marketplace/gpu

**Comment by cryptoni9n** [comment]
> Spellbound
> 
> https://ordinals.com/children/451accbce30177c23a8cd42ab85131312a449359ada3c054aa147b054136fa3bi0
> 
> Utilizes recursion, delegation, metadata, parent/child, postage, location, compression. Background here https://x.com/lifofifo/status/1767059753555853656

added under provenance

**Comment by cryptoni9n** [comment]
> Actully GPU Ordinals by Anarkoic is another cool collection ,here are some self-intro from the Magic Eden:
> 
> GPU Ordinals by Anarkoic is the first ever WebGPU Ordinals collection. WebGPU is the new technology that provides high-throughput use of GPU's for graphics and even computation, & can enable high-performance Metaverses and LLM AI's on Bitcoin. This immersive, interactive 3D collection also features Grandparent-Grandchild Inscription, Spatial Audio, Real-time Block Streaming & introduces Super-Recursion. Max Supply 2100.
> 
> https://magiceden.io/ordinals/marketplace/gpu

I agree there are a lot of cool things about this collection, but I'm not sure where best to place this as an example.  Perhaps @Anarkoic can provide some information here about 'super recursion' so I don't miscommunicate?

**Comment by Anarkoic** [comment]
> > Actully GPU Ordinals by Anarkoic is another cool collection ,here are some self-intro from the Magic Eden:
> > GPU Ordinals by Anarkoic is the first ever WebGPU Ordinals collection. WebGPU is the new technology that provides high-throughput use of GPU's for graphics and even computation, & can enable high-performance Metaverses and LLM AI's on Bitcoin. This immersive, interactive 3D collection also features Grandparent-Grandchild Inscription, Spatial Audio, Real-time Block Streaming & introduces Super-Recursion. Max Supply 2100.
> > https://magiceden.io/ordinals/marketplace/gpu
> 
> I agree there are a lot of cool things about this collection, but I'm not sure where best to place this as an example. Perhaps @Anarkoic can provide some information here about 'super recursion' so I don't miscommunicate?

Hi, thanks Super-Recursion leverages "Recursion" and mutates it. Content is pulled in from the recursion url and then changed before rendering. In this case the large WebGPU Three.js inscription used by the collection is leveraged, and changes developed post-inscription are used without having to reinscribe the large dependent inscription.

**Comment by cryptoni9n** [comment]
> Hi, thanks Super-Recursion leverages "Recursion" and mutates it. Content is pulled in from the recursion url and then changed before rendering. In this case the large WebGPU Three.js inscription used by the collection is leveraged, and changes developed post-inscription are used without having to reinscribe the large dependent inscription.

added to recursion.md

**Review by casey — CHANGES_REQUESTED** [review]
This all looks good, but I think I would prefer all the examples to be in a single section of the documentation, probably `docs/src/inscriptions/examples.md`.

If someone is interested in a particular feature, then having the example in the section for that feature is useful, but if they're just interested in features of ordinals generally, then being able to browse them all on a dedicated page is nice.

**Comment by mutabletrees** [comment]
You can't not include the first recursive inscription in the recursive section.  Just saying.  
https://ordinals.com/inscription/bf637552f3758e20f733d1f250cbea7cbbb4bbf157a8d4a9c26132950a383415i0

File: docs/src/inscriptions/examples.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+Inscription Examples
+=========
+
+### Delegate
+
+* The Oscillations * collection utilizes delegation, provenance, recursion, sat endpoint, and detects the kind of sat that each piece is inscribed on (sattribute-aware). Each piece is a delegate of [this inscription](https://ordinals.com/inscription/52b4ea10c2518c954c73594e403ccfb2d50044f5a3b09a224dfa3bf06dd1d499i0).
```
Comment (lifofifoX) [inline docs/src/inscriptions/examples.md:6]: First ever delegation inscription (afaik) https://ordinals.com/inscription/626127511953479e1933a448f020c76acd9086efe2f2b3f5e492c0c85d2ce051i0

Comment (cryptoni9n) [reply docs/src/inscriptions/examples.md:7]: agree - added

**Comment by cryptoni9n** [comment]
keep getting these weird lint failures on src/index.rs, which is not a part of this PR.

>    Checking update-contributors v0.0.0 (/home/runner/work/ord/ord/crates/update-contributors)
error: the borrowed expression implements the required traits
    --> src/index.rs:4518:53
     |
4518 |             inscription("text/plain;charset=utf-8", &format!("hello {}", i)).to_witness(),
     |                                                     ^^^^^^^^^^^^^^^^^^^^^^^ help: change this to: `format!("hello {}", i)`
     |
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_borrows_for_generic_args
     = note: `-D clippy::needless-borrows-for-generic-args` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(clippy::needless_borrows_for_generic_args)]`

>error: could not compile `ord` (lib test) due to 1 previous error
Error: Process completed with exit code 101.

going to close/reopen this PR to re-initiate the checks.

**Comment by cryptoni9n** [comment]
getting identical failures 3 times in a row now.  @raphjaph / @casey - would one of you please take a look and let me know what you think is going wrong here?

**Comment by casey** [comment]
@cryptoni9n This was due to a recent rust update which caused a clippy lint to fail. Just merged #3937 to fix it.

**Review by raphjaph — APPROVED** [review]
These are great, thanks for aggregating these!

**Pull request merged**